command/agent.go

```diff
@@ -193,81 +201,29 @@ func (c *AgentCommand) Run(args []string) int {
 		return 1
 	}
 
-	config := agentConfig.NewConfig()
-
-	for _, configPath := range c.flagConfigs {
-		configFromPath, err := agentConfig.LoadConfig(configPath)
-		if err != nil {
-			c.UI.Error(fmt.Sprintf("Error loading configuration from %s: %s", configPath, err))
-			return 1
-		}
-		config = config.Merge(configFromPath)
-	}
-
-	err := config.ValidateConfig()
-	if err != nil {
-		c.UI.Error(fmt.Sprintf("Error loading configuration: %s", err))
-		return 1
-	}
-
-	if config.AutoAuth == nil {
-		c.UI.Info("No auto_auth block found in config, not starting automatic authentication feature")
-	}
-
-	c.updateConfig(f, config)
-
-	// Parse all the log related config
-	logLevel, err := logging.ParseLogLevel(config.LogLevel)
-	if err != nil {
-		c.UI.Error(err.Error())
-		return 1
-	}
-
-	logFormat, err := logging.ParseLogFormat(config.LogFormat)
+	cfg, err := c.loadConfig(c.flagConfigs)
 	if err != nil {
-		c.UI.Error(err.Error())
+		c.outputErrors(err)
 		return 1
 	}
 
-	logRotateDuration, err := parseutil.ParseDurationSecond(config.LogRotateDuration)
-	if err != nil {
-		c.UI.Error(err.Error())
-		return 1
+	if cfg.AutoAuth == nil {
+		c.UI.Info("No auto_auth block found in config, the automatic authentication feature will not be started")
 	}
 
-	logRotateBytes, err := parseutil.ParseInt(config.LogRotateBytes)
-	if err != nil {
-		c.UI.Error(err.Error())
-		return 1
-	}
+	c.updateConfig(f, cfg) // This only needs to happen on start-up to aggregate config from flags and env vars
+	c.config = cfg
 
-	logRotateMaxFiles, err := parseutil.ParseInt(config.LogRotateMaxFiles)
+	l, err := c.newLogger()
 	if err != nil {
-		c.UI.Error(err.Error())
+		c.outputErrors(err)
 		return 1
 	}
-
-	logCfg := &logging.LogConfig{
-		Name:              "vault-agent",
-		LogLevel:          logLevel,
-		LogFormat:         logFormat,
-		LogFilePath:       config.LogFile,
-		LogRotateDuration: logRotateDuration,
-		LogRotateBytes:    int(logRotateBytes),
-		LogRotateMaxFiles: int(logRotateMaxFiles),
-	}
-
-	l, err := logging.Setup(logCfg, c.logWriter)
-	if err != nil {
-		c.UI.Error(err.Error())
-		return 1
-	}
-
 	c.logger = l
 
 	infoKeys := make([]string, 0, 10)
 	info := make(map[string]string)
-	info["log level"] = config.LogLevel
+	info["log level"] = c.config.LogLevel
```

Comment:
This diff is a lot harder to read than it needs to be, because of all the changes from `config` to `c.config` in this function.  Could you take pity on your reviewers please, and revert that change temporarily?  Doesn't need to change any coded behaviour, c.config is a pointer anyway, so we can continue to populate that field and reference it in other methods.  After review you can introduce a new commit that brings back these 70+ c.config changes.

Reply:
Sorry about that! Whilst the damage to your eyesight may now be irreversible, I'll revert the change now and push it up. I hope you can grant me forgiveness. :)

Reply:
Oh I made the change locally already, but I figured there may be other reviewers like Violet, no reason for us all to have to do it when it could be done in one place.  No apology needed, ty for accommodating me.